### ISSUE TITLE
hard to reproduce the results on VGG16-bn for Cifar10

### ISSUE THREAD

**Issue by Miocio-nora** [tianyic/only_train_once#37]
We use RTX4090 for training. With the tutorial, we trained over 15 times with diffirent seeds setting (including many times with the initial seed setting 42). However, these experiments results are close, with highest result reached best 93.0% ACC, but is still lower than 93.2% from the lecture. 
here are the experiments env:
numpy  1.23.4      python 3.8.15
pytorch 1.13.0      cuda 11.6.2
only_train_once =2.0.16

Can i have any suggestions and env detials to continue reproducing this experiment.
Should i further repeat the experiments with the seed 42, or continuing try other seeds?
![ACC with several runs](https://github.com/tianyic/only_train_once/assets/108513100/879b4ab4-a070-429e-8aec-e4d0ece9694f)

**Comment by tianyic** [comment]
We are using NVIDIA TITAN RTX when creating the tutorial as our A100 being occupied for larger experiments. 

Your results look normal. On our end, the VGG16 experiments fluctuates more than other experiments as well. The highest accuracy about 95% parameter reductions is 93.4%. Under the same parameter settings and even the same random seed, the accuracy could fluctuate about 0.5% or a bit more. 

The **root cause** might be that the compression on VGG16-BN CIFAR10 has been **closed to fully exploitations**, thereby **no tolerated redundant structures may exist**. The results then become sensitive especially.  The hardware and software discrepancy and randomness may result in some indecipherable fluctuations. Furthermore, the straight structures like VGG16 may also amplify the above. 

We share the training log here

[vgg16bn_cifar10_27.csv](https://github.com/tianyic/only_train_once/files/12531651/vgg16bn_cifar10_27.csv)

 that could reach 93.4% and agree that reaching over 93.0% accuracy may be fortune-dependent regarding convergence over the objective landscape. (We were also suffering from beating previous results in the past, as shown that we set more sophisticated hyper-parameters.)

To more reliably get higher accuracy, you could try to reduce the target group sparsity to 78%, more random seeds and different GPUs (fortune-dependent), other learning rate scheduler or a reasonable pre-training checkpoint.

**Comment by Miocio-nora** [comment]
Thanks for those shared detials and suggestions. We have succesfully reached at best 93.3% accuracy by now, by repeating the experiments with the seed 42 over many times. We will also try to get more stable results by following the suggestions above.
